code/game/objects/items/storage/boxes.dm

```diff
@@ -415,10 +413,19 @@
 	new /obj/item/reagent_container/food/snacks/packaged_meal(src, snack)
 	new /obj/item/reagent_container/food/snacks/packaged_meal(src, desert)
 
+
 /obj/item/storage/box/MRE/remove_from_storage()
 	. = ..()
-	if(. && !contents.len && !gc_destroyed)
-		qdel(src)
+	if(length(contents))
+		return
+
+	if(!istype(loc, /obj/item/storage))
```

Comment:
Where is this being called from out of a storage?

Reply:
The issue is our super smart deletion subsystem that calls destroy after a delay and not immediatelly

Reply:
Also remove_from_storage is for removing things FROM THIS OBJECT